### DISCUSSION TITLE
Strings for openAI models?

### DISCUSSION THREAD

**Discussion by proteus-dev** [DeabLabs/cannoli#11] [Q&A] [answered]
Hello again,

I just tried to parse a larger document through a cannoli and I reached the rate limit for the chosen model, which was gpt-3.5-turbo. So I thought I just switch to gpt-3.5-turbo-16k and problem solved. However, I fail at configuring the cannoli: 

- what is the right string for gpt-3.5-turbo-16k?

on a similar note: 
- what are temperature ranges (1 to 2?)
- are the ranges used in the plugin config and via configuration arrows the same? (I guess so but you never know...)

Thanks! :)

**Comment by blindmansion** [comment, marked as answer]
Yes, that's the correct string. You can use any of the strings found in the models table of the openai docs (https://platform.openai.com/docs/models/gpt-3-5). Note that choice and field arrows only work when the model is capable of function-calling (also be found on the docs)

I had thought the temperature range is between 0-2, but the docs are saying it's 0-1, so I'm not exactly sure. Might have to do some trial and error to figure that out.

The settings panel setting will be overridden by the config arrow if you have one.

**Comment by proteus-dev** [comment]
Thanks for the temperatures info!
could you tell me what I am doing wrong regarding the models (see below)?

<img width="348" alt="image" src="https://github.com/DeabLabs/cannoli/assets/121372604/69af228b-f333-485b-b749-5d8af350c656">
<img width="578" alt="image" src="https://github.com/DeabLabs/cannoli/assets/121372604/03d3b473-5538-4d84-ac3e-94009d508d38">
<img width="1220" alt="image" src="https://github.com/DeabLabs/cannoli/assets/121372604/dab3d40a-833e-4a8d-a7da-9a03fa0ac1e6">

**Reply by proteus-dev** [reply]
Just to be sure I am not just using the wrong table, I also tried it with gpt-3.5-turbo-1106

<img width="529" alt="image" src="https://github.com/DeabLabs/cannoli/assets/121372604/3c07fc00-6525-4256-856e-a9cd2021292c">

**Reply by blindmansion** [reply]
Yes this is a bug that's been raised by others, you're using the strings correctly. I still need to get in and write a fix.

Sorry it's been some wait on these bugs, I've had trouble finding time the last few weeks.

**Reply by proteus-dev** [reply]
No problem. Wasn't aware this is an already found bug.

**Reply by blindmansion** [reply]
Finally fixed this! The newest release has support for all the new openai language models.
